File: semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/ScipRole.java
package com.sourcegraph.semanticdb_javac;

/**
 * Minimal role enum used by {@link ScipVisitor} when emitting SCIP occurrences. Mirrors the subset
 * of {@code Semanticdb.SymbolOccurrence.Role} that the direct-to-SCIP visitor cares about.
 */
enum ScipRole {
  DEFINITION,
  REFERENCE,
  SYNTHETIC_DEFINITION
}

File: semanticdb-kotlinc/src/main/kotlin/com/sourcegraph/semanticdb_kotlinc/ScipRole.kt
package com.sourcegraph.semanticdb_kotlinc

/**
 * Minimal role enum used by the direct-to-SCIP visitor and builder. Mirrors the subset of
 * `Semanticdb.SymbolOccurrence.Role` that the Kotlin plug-in needs after the SemanticDB protobuf
 * dependency has been removed.
 */
enum class ScipRole {
    DEFINITION,
    REFERENCE
}
